spec/haskell/src/SEL4/Object/TCB.lhs

```diff
@@ -341,17 +341,31 @@ This is to ensure that the source capability is not made invalid by the deletion
 >     vRoot <- if isValidVTableRoot vRootCap'
 >         then return (vRootCap', vRootSlot)
 >         else throw IllegalOperation
+>     return $ ThreadControlCaps {
+>         tcCapsTarget = capTCBPtr cap,
+>         tcCapsSlot = slot,
+>         tcCapsFaultHandler = Nothing,
+>         tcCapsTimeoutHandler = Nothing,
+>         tcCapsCRoot = Just cRoot,
+>         tcCapsVRoot = Just vRoot,
+>         tcCapsBuffer = Nothing }
+> decodeCVSpace _ _ _ _ = throw TruncatedMessage
+
+> decodeSetSpace :: [Word] -> Capability -> PPtr CTE ->
+>         [(Capability, PPtr CTE)] -> KernelF SyscallError TCBInvocation
+> decodeSetSpace (cRootData:vRootData:_) cap slot (cRootArg:vRootArg:fhArg:_) = do
```

Comment:
The C code has the `fhArg` in the original first extra cap slot. I can see a pretty strong argument for why this order is preferred, so that it's more consistent with the non MCS code, but is there a corresponding change being made to the C?

Reply:
Ah, good catch. I suppose for this part, I was happily following along with `decode_set_space` in the abstract. But I think that's wrong. So I'll change both

Reply:
That is a good catch. Thanks Corey. I can see that C has the FH first, not sure how abstract came to have that order.